### PR TITLE
[SPARK-7403][WebUI] Link URL in objects on Timeline View is wrong in case of running on YARN 

### PR DESCRIPTION
**Pull request by sarutak**
When we use Spark on YARN and have AllJobPage via ResourceManager's proxy, the link URL in objects which represent each job on timeline view is wrong.

In timeline-view.js, the link is generated as follows.

```
window.location.href = "job/?id=" + getJobId(this);
```

This assumes the URL displayed on the web browser ends with "jobs/" but when we access AllJobPage via the proxy, the url displayed does not end with "jobs/" 

The proxy doesn't return status code 301 or 302 so the url displayed still indicates the base url, not "/jobs" even though displaying AllJobPages.

### CODE REVIEW

**Comment by andrewor14** [comment]
retest this please

File: core/src/main/resources/org/apache/spark/ui/static/timeline-view.css [hunk 1]
```diff
@@ -19,7 +19,8 @@ div#application-timeline, div#job-timeline {
   margin-bottom: 30px;
 }
 
-#application-timeline div.legend-area {
+#application-timeline div.legend-area ,
```
Comment (andrewor14) [inline core/src/main/resources/org/apache/spark/ui/static/timeline-view.css]: no space before ,

**Comment by andrewor14** [comment]
Thanks @sarutak LGTM.

**Comment by andrewor14** [comment]
By the way, it's a separate issue but when I open the JS console I see some errors with `viz.js`
```
Failed to load resource: the server responded with a status of 404 (Not Found)
http://localhost:5050/static/dist/vis.js
```
Do you know what's wrong?

**Comment by sarutak** [comment]
@andrewor14 Thanks for letting me know another issue. I found the reason is `vis.js` (not vis.min.js) is absent. `vis.js` is designated in `vis.map` which is a file for Source Maps.

Source Maps is used for symbol resolution for abbreviated symbols in *.min.js (http://www.html5rocks.com/en/tutorials/developertools/sourcemaps/).

`vis.map` and `vis.js` are only downloaded when we use debug tools embedded in browsers.

File: .rat-excludes [hunk 2]
```diff
@@ -35,6 +35,7 @@ dagre-d3.min.js
 graphlib-dot.min.js
 sorttable.js
 vis.min.js
+vis.js
```
Comment (srowen) [inline .rat-excludes]: Why is this needed now?

Comment (sarutak) [inline .rat-excludes]: It's because `vis.js` is referred by `vis.map` and `vis.map` is referred by vis.min.js.
`vis.js` is only downloaded when Source Maps is enabled for debug.
Firefox or Chrome is supported SourceMaps and browsers which don't support Source Maps just ignore the reference directive in vis.min.js.

Comment (srowen) [inline .rat-excludes]: Why does that affect the Spark source tree though? Maybe dumb question

**Comment by zsxwing** [comment]
@sarutak could you also add `vis` to the LICENSE file, please?

**Comment by zsxwing** [comment]
@sarutak this issue only happens in the `index` page. In other pages, we can use the relative path. Right?

File: .rat-excludes (hunk 2 above)
Comment (andrewor14) [inline .rat-excludes]: I think it's because we can't expect the cluster to have network access to the external world so we need to bundle it.

But @sarutak what do we use `vis.map` for? The timeline seems to work fine even without `vis.js`. What are the specific UI differences?

Comment (zsxwing) [inline .rat-excludes]: I think it's just for debuging. See https://developer.chrome.com/devtools/docs/javascript-debugging

Comment (zsxwing) [inline .rat-excludes]: Search "Using Source Maps" in this page

Comment (zsxwing) [inline .rat-excludes]: But I think we don't need to include "vis.js" in Spark since it's just for debuging. The developers can add them by themselves locally.

Comment (andrewor14) [inline .rat-excludes]: In that case I would prefer to leave it out if possible. This is bringing in a 30k lines file into our repo, not to mention that it slows down the page load speed.

**Comment by andrewor14** [comment]
@sarutak sorry I wasn't being clear. I would prefer that this patch only address the YARN issue so we can merge it more easily. We should address the missing `vis.js` error separately (either by removing usages of it, which I prefer, or pulling in the needed dependency).

**Comment by sarutak** [comment]
I think Source Maps doesn't affect page load speed because non-compressed `vis.js` is only downloaded when we debug (using developer tools of Firefox or like that)  but there is some reason in what you says.

Anyway, I'll remove `vis.js` from my changes for now.
Tnanks @andrewor14 ,@srowen and @zsxwing !

**Comment by sarutak** [comment]
@zsxwing Yes, I think it happens only `index` page of Web UI via ResourceManager's proxy (HistoryServer don't have this issue).

**Comment by sarutak** [comment]
retest this please.
